### PR TITLE
Fixes auto-succeeding pester tests in AboutComparison and AboutStrings

### PR DESCRIPTION
**Pull request by EVWorth**
﻿# PR Summary

This PR fixes pester tests from auto-succeeding.

This PR Resolves Issues #212 and #213 

I tested these changes to make sure they initially fail the pester tests successfully

## Context

The solution for the AboutComparison issues came from: https://github.com/vexx32/PSKoans/issues/212#issuecomment-507712857

The solution for the AboutString issues came from: https://github.com/vexx32/PSKoans/issues/213#issuecomment-507713959

## Changes

In AboutStrings:
- I replaced text in $String1 on line 105 that was causing the test to succeed without user input. 

In AboutComparison.Koans.ps1
-  I replaced 4 instances of __ with $__, where the pester tests were succeeding without user input, because __ was evaluating to $true which completed the test. 

## Checklist

- [X] Pull Request has a meaningful title.
- [X] Summarised changes.
- [X] Pull Request is ready to merge & is not WIP.
- [X] Added tests / only testable interactively.
  - Make sure you add a new test if old tests do not effectively test the code changed.
- [X] Added documentation / opened issue to track adding documentation at a later date.

### CODE REVIEW

**Comment by vexx32** [comment]
Looks good! Can we use $____ rather than $__ (four underscores instead of two) just so it's very clear we're not trying to reference $_?

**Comment by EVWorth** [comment]
Good point/idea!

Done!

File: PSKoans/Koans/Foundations/AboutStrings.Koans.ps1 [hunk 1]
```diff
@@ -102,7 +102,7 @@ Describe 'Strings' {
 
         It 'adds strings together' {
             # Two become one.
-            $String1 = 'This string'
+            $String1 = '__'
```
Comment (vexx32) [inline PSKoans/Koans/Foundations/AboutStrings.Koans.ps1]:
```suggestion
            $String1 = '_____'
```

Been trying to move towards using long-blanks for strings as well, just to make it more clear that we're expecting more than a couple of characters. 🙂

**Comment by EVWorth** [comment]
I like the longer blanks. They stand out more which is always a good thing. Do you have a preference on standardizing the number of underscores?

perhaps 4 of each? ____ for int/double, `"____"` for string, and $____ for Boolean? 

I know that the Foundations Koans could use a good freshening up in that regard. sounds like a project...  😄

**Comment by vexx32** [comment]
@EVWorth yeah it's changed over time as I realize this might be a bit clearer for various reasons. I'd say.... if we're expecting: 
* Numeric input, go with `__` 
* Strings, go with `'____'`
* Variables, go with `$____`
* Command names or strings in command arguments, go with `____` unless the string you expect would normally require quotes to enter it at a command line.

Of course, with strings it's good to think about whether the string you expect might make use of interpolation and should thus use double quotes instead of single quotes. 🙂

**Comment by EVWorth** [comment]
Ok awesome. I'll use that going forward. I was thinking about the ' ' vs "" thing earlier. I think it makes sense to differentiate between those as well.

**Comment by vexx32** [comment]
🤔 Sigh, I need to add a contributing style guide lmao

**Comment by EVWorth** [comment]
If you do I'll contribute to it and ask you 50 billion questions you don't want to have to answer ;)

**Comment by vexx32** [comment]
I don't mind answering haha, it's all good. Just something I should do. I guess I'll make an issue so I don't forget. :)

**Pull request merged**